scanner/services/matcher.go

```diff
@@ -146,3 +148,22 @@ func (s *matcherService) RegisterServiceHandler(_ context.Context, _ *runtime.Se
 	// Currently we do not set up gRPC gateway for the matcher.
 	return nil
 }
+
+func (s *matcherService) notes(ctx context.Context, vr *v4.VulnerabilityReport) []v4.VulnerabilityReport_Note {
+	dists := s.matcher.GetKnownDistributions(ctx)
+	if len(dists) == 0 {
+		return []v4.VulnerabilityReport_Note{v4.VulnerabilityReport_NOTE_OS_VULNERABILITIES_UNAVAILABLE}
+	}
+	dist := vr.Contents.Distributions[0]
+	dID := dist.GetDid()
+	versionID := dist.GetVersionId()
+	known := slices.ContainsFunc(dists, func(dist claircore.Distribution) bool {
```

Comment:
Breaking out the experimental features I see 😉

Reply:
not experimental in go1.21, but we can't use that yet :(